src/go/k8s/apis/redpanda/v1alpha1/console_types.go

```diff
@@ -0,0 +1,189 @@
+
+	BasePath string `json:"basePath,omitempty" yaml:"basePath,omitempty"`
+
+	// +kubebuilder:default=true
+	SetBasePathFromXForwardedPrefix bool `json:"setBasePathFromXForwardedPrefix,omitempty" yaml:"setBasePathFromXForwardedPrefix,omitempty"`
+
+	// +kubebuilder:default=true
+	StripPrefix bool `json:"stripPrefix,omitempty" yaml:"stripPrefix,omitempty"`
+}
+
+// Schema defines configurable fields for Schema Registry
+type Schema struct {
+	Enabled bool `json:"enabled"`
+}
+
+// Deployment defines configurable fields for the Console Deployment resource
+type Deployment struct {
+	Image string `json:"image"`
+
+	// +kubebuilder:default=1
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// +kubebuilder:default=0
+	MaxUnavailable int32 `json:"maxUnavailable,omitempty"`
+
+	// +kubebuilder:default=1
+	MaxSurge int32 `json:"maxSurge,omitempty"`
+}
+
+// Connect defines configurable fields for Kafka Connect
+type Connect struct {
+	// +optional
+	Enabled bool `json:"enabled"`
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=duration
+	// +kubebuilder:default="15s"
+	ConnectTimeout *metav1.Duration `json:"connectTimeout,omitempty"`
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=duration
+	// +kubebuilder:default="60s"
+	ReadTimeout *metav1.Duration `json:"readTimeout,omitempty"`
+
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Format=duration
+	// +kubebuilder:default="6s"
+	RequestTimeout *metav1.Duration `json:"requestTimeout,omitempty"`
+
+	Clusters []ConnectCluster `json:"clusters,omitempty"`
+}
+
+// ConnectCluster defines configurable fields for the Kafka Connect cluster
+type ConnectCluster struct {
+	Name string `json:"name"`
+	URL  string `json:"url"`
+
+	// TLS configures mTLS auth
+	TLS *ConnectClusterTLS `json:"tls,omitempty"`
+
+	// BasicAuthRef configures basic auth credentials referenced by Secret
+	// Expects to have keys "username", "password"
+	BasicAuthRef *corev1.ObjectReference `json:"basicAuthRef,omitempty"`
+
+	// TokenRef configures token header auth referenced by Secret
+	// Expects to have key "token"
+	TokenRef *corev1.ObjectReference `json:"tokenRef,omitempty"`
+}
+
+// ConnectClusterTLS defines TLS certificates for the Kafka Connect cluster
+type ConnectClusterTLS struct {
+	Enabled bool `json:"enabled,omitempty"`
+
+	// SecretKeyRef configures certificate used for mTLS auth referenced by Secret
+	// Expects to have keys "tls.crt", "tls.key", "ca.crt"
+	SecretKeyRef *corev1.ObjectReference `json:"secretKeyRef,omitempty"`
+
+	InsecureSkipTLSVerify bool `json:"insecureSkipTlsVerify,omitempty"`
+}
+
+// ConsoleStatus defines the observed state of Console
+type ConsoleStatus struct {
+	// The ConfigMap used by Console
+	// This is used to pass the ConfigMap used to mount in the Deployment Resource since Ensure() only returns error
+	ConfigMapRef *corev1.ObjectReference `json:"configMapRef,omitempty"`
```

Comment:
why do we need to surface this in the status? isn't this internal detail?

Reply:
This tracks the current used `ConfigMap` by Console. Updating a field in Console spec will trigger a reconcile and update the child ConfigMap. But this will not trigger an update of the Pods (e.g. changing listenPort in the spec will not change the actual port listed to by Pod). Instead, we set ConfigMap `immutable=true` and create new ConfigMap every spec change. More on the recording with Nicola :)